Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Pass.cpp

```diff
@@ -469,95 +479,138 @@ namespace AZ
+
+                foundPass = true;   // Using the "Pipeline" keyword, no need to continue searching for passes
+
+                if (m_pipeline)
                 {
-                    foundPass = true;
+                    const PipelineGlobalBinding* globalBinding = m_pipeline->GetPipelineGlobalConnection(connectedSlotName);
+                    if (globalBinding)
+                    {
+                        connectedBinding = m_pipeline->GetPipelineGlobalConnection(connectedSlotName)->m_binding;
+                    }
+                    if (!connectedBinding)
+                    {
+                        attachment = m_pipeline->GetPipelineGlobalAttachment(connectedSlotName);
+                    }
 
-                    // Get the connected binding from the parent
-                    connectedBinding = m_parent->FindAttachmentBinding(connectedSlotName);
+                    AZ_RPI_PASS_ERROR(connectedBinding || attachment, "Pass::ProcessConnection - Pass [%s] cannot find attachment or binding named [%s] on pipeline.",
+                        m_path.GetCStr(),
+                        connectedSlotName.GetCStr());
+                }
+            }
 
-                    bool slotTypeMismatch = connectedBinding != nullptr &&
-                        connectedBinding->m_slotType != localBinding->m_slotType &&
-                        connectedBinding->m_slotType != PassSlotType::InputOutput &&
-                        localBinding->m_slotType != PassSlotType::InputOutput;
+            // -- Search Parent & Siblings --
 
-                    if (slotTypeMismatch)
+            // The (connectedPassName != m_name) avoids edge case where parent pass has child pass of same name.
+            // In this case, parent pass would ask it's parent pass for a sibling with the given name and get a pointer to itself.
+            // It would then try to connect to itself, which is obviously not the intention of the user
+            if (!foundPass && m_parent && connectedPassName != m_name)
+            {
+                if (connectedPassName == PassNameParent)
+                {
+                    foundPass = true;
+                    connectedBinding = m_parent->FindAttachmentBinding(connectedSlotName);
+                    if (!connectedBinding)
                     {
-                        AZ_RPI_PASS_ERROR(false, "Pass::ProcessConnection - When connecting to a parent slot, both slots must be of the same type (or one must be InputOutput)");
-                        connectedBinding = nullptr;
+                        attachment = m_parent->FindOwnedAttachment(connectedSlotName);
+                    }
+                    else
+                    {
+                        slotTypeMismatch = connectedBinding->m_slotType != localBinding->m_slotType &&
+                            connectedBinding->m_slotType != PassSlotType::InputOutput &&
+                            localBinding->m_slotType != PassSlotType::InputOutput;
```

Comment:
Can all the slotTypeMismatch check be converted to an lambda function?

Reply:
Differs for whether the connection is to a parent or neighbor.